### ISSUE TITLE
[docs] TrainingArguments default label_names is not what is described in the documentation

### ISSUE THREAD

**Issue by fredsensibill** [huggingface/transformers#21287]
### System Info

- `transformers` version: 4.25.1
- Platform: macOS-12.6.1-arm64-arm-64bit
- Python version: 3.8.15
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 1.13.1 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: No

### Who can help?

@sgugger, @stevhliu and @MKhalusova

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

1. Create a model with a `forward` that has more than one label. For example:

```
 def forward(
        self,
        input_ids,
        bbox,
        attention_mask,
        token_type_ids,
        labels,
        reference_labels
)
```

2. Create a trainer for your model with `trainer = Trainer(model, ...)`. Make sure to not set `label_names` and let it default.
3. Check `trainer.label_names` and see that it returns `["labels", "reference_labels"]`

### Expected behavior

[The documentation](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.Seq2SeqTrainingArguments.label_names) states that:

> Will eventually default to ["labels"] except if the model used is one of the XxxForQuestionAnswering in which case it will default to ["start_positions", "end_positions"].

[This PR](https://github.com/huggingface/transformers/pull/16526) changed the behaviour that the documentation describes.

**Comment by sgugger** [comment]
Indeed. Do you want to open a PR to fix the documentation?
